src/chrome/content/rules/Sohu.com-mixcontent.xml

```diff
@@ -0,0 +1,45 @@
+<!--
+	Other Sohu rulesets:
+		- Sohu.com.xml
+		- Sohu.net.xml
+-->
+
+<ruleset name="Sohu.com-mixedcontent">
+
+	<target host="www.sohu.com" />
+	<exclusion pattern="^http://www\.sohu\.com/$" />
+	<rule from="^http://www\.sohu\.com/sohuflash_1\.js"
+			to="https://www.sohu.com/sohuflash_1.js" />
+	<rule from="^http://www\.sohu\.com/upload/"
+			to="https://www.sohu.com/upload/" />
+		<test url="http://www.sohu.com/sohuflash_1.js" />
+		<test url="http://www.sohu.com/upload/style/style150302.css" />
+		<test url="http://www.sohu.com/upload/images9_18/bg.png" />
+		<test url="http://www.sohu.com/upload/images20151228/bg01.gif" />
+		<test url="http://www.sohu.com/upload/images/server/icon08.jpg" />
+		
+	<target host="news.sohu.com" />
+	<exclusion pattern="^http://news\.sohu\.com/$" />
+	<rule from="^http://news\.sohu\.com/upload/"
+			to="https://news.sohu.com/upload/" />
+		<test url="http://news.sohu.com/upload/style.css" />
+		<test url="http://news.sohu.com/upload/images/pic09.jpg" />
+		<test url="http://news.sohu.com/upload/images/flash.swf" />
+
+	<target host="tv.sohu.com" />
+	<exclusion pattern="^http://tv\.sohu\.com/$" />
+	<rule from="^http://tv\.sohu\.com/([\w/-]+)\.(jpg|js|png|swf)"
+			to="https://tv.sohu.com/$1.$2" />
+		<test url="http://tv.sohu.com/upload/space/skin/imgs/avatar_s.jpg" />
+		<test url="http://tv.sohu.com/frag/vrs_inc/phb_variety_day_10_simple.js" />
+		<test url="http://tv.sohu.com/upload/Trace/wrating.js" />
+		<test url="http://tv.sohu.com/upload/static/global/skin/images/icon_merge.png" />
+		<test url="http://tv.sohu.com/upload/swf/20161102/Main.swf" />
+
+	<target host="push.my.tv.sohu.com" />
+	<exclusion pattern="^http://push\.my\.tv\.sohu\.com/$" />
+	<rule from="^http://push\.my\.tv\.sohu\.com/(\S+)\.json"
+			to="https://push.my.tv.sohu.com/$1.json" />
+		<test url="http://push.my.tv.sohu.com/getwebpush_sohuHD.globalNewPushTrigger.json" />
```

Comment:
Judging from the ruleset [test](https://travis-ci.org/EFForg/https-everywhere/jobs/175450396), I think you need to add one more link with `.json`.